### ISSUE TITLE
'anzu-replace-threshold' is ignored for 'anzu-query-replace' 

### ISSUE THREAD

**Issue by tttuuu888** [emacsorphanage/anzu#117]
Hello. Thanks a lot for this great package.

Here is reproducing steps:
` $ emacs -Q`
```
(package-initialize)
(require 'anzu)
(global-anzu-mode 1)
```

and open the latest `anzu.el` file.

`C-s` `a` shows (1/1000+) as default `anzu-search-threshold` is set to 1000.
However `M-x` - `anzu-query-replace` `a` shows (1650 replace) although `anzu-replace-threshold` is set to default 1000 too.
I also found same result with `anzu-query-replace-regexp`.

Thanks.

**Comment by syohex** [comment]
`anzu-replace-threshold` does not work same as `anzu-search-threshold`. `anzu-replace-threshold` works for the number of overlays, not total count like `anzu-replace-threshold`. Too many overlays make emacs slow, so I introduced this variable.

**Comment by tttuuu888** [comment]
Thank you for the explanation.

Actually, I opened this issue because I found `anzu-query-replace` is quite slow to prepare total count. Mostly it is okay but if the count exceed 100000, I see `anzu-query-replace` hangs for seconds to only prepare the total count before the real processing.

So, I'm wondering if there is any chance to update `anzu-query-replace` to work the way `isearch` with `anzu-search-threshold`.
